pkgs/servers/teleport/default.nix

```diff
@@ -97,35 +80,29 @@ buildGoModule rec {
   # Reduce closure size for client machines
   outputs = [ "out" "client" ];
 
-  preBuild =
-    let rustDeps = symlinkJoin {
-      name = "teleport-rust-deps";
-      paths = lib.optional withRdpClient rdpClient
-        ++ lib.optional withRoleTester roleTester;
-    };
-    in
-    ''
-      mkdir -p build
-      echo "making webassets"
-      cp -r ${webassets}/* webassets/
-      make lib/web/build/webassets
-
-      cp -r ${rustDeps}/. .
-    '';
+  preBuild = ''
+    mkdir -p build
+    echo "making webassets"
+    cp -r ${webassets}/* webassets/
+    make lib/web/build/webassets
+  '' + lib.optionals withRdpClient ''
```

Comment:
I have a suspicion that this works by accident due to `lib.optionals` not doing any type checks

https://github.com/NixOS/nixpkgs/blob/55eb08677944821cb5b992e906c0cbecbe027bcc/lib/lists.nix#L267-L271

However, seems like it will fail when `withRdpClient == false`, because in that case you're trying to `string + list`.

`lib.optionalString` should be the way to go here.